### ISSUE TITLE
Testable examples must have an `output` comment

### ISSUE THREAD

**Issue by jamietanna** [golangci/golangci-lint#3084]
### Your feature request related to a problem? Please describe.

When writing [testable examples](https://pkg.go.dev/testing#hdr-Examples), the following code is expected:

```go
func ExampleHello() {
    fmt.Println("hello")
    // Output: hello
}
```

Or a lowercase `output`:

```go
func ExampleHello() {
    fmt.Println("hello")
    // output: hello
}
```

However, when writing the following, without an `Output` comment:

```go
func ExampleHello() {
    fmt.Println("hello")
}
```

Then the test compiles correctly, but is not picked up by `go test`.

We should add a rule which flags when an `output` comment is missing.

### Describe the solution you'd like.

Add a new rule which flags when a testable example - a test that has the test method prefix `Example` - does not contain a comment with an `Output:`

### Describe alternatives you've considered.

This may not be fit to come into golangci-lint per se, but into one of the other linters.

**Comment by maratori** [comment]
At first glance it's a good idea.
But if I would like to create an example without `output`, I can't use `//nolint` and will have to use `exclude-rules` in `.golangci.yml`.
That's because `//nolint` comment will be added to godoc.
```go
func Example_sameLine() { // nolint:testableexamples
	panic("xxx")
}

// nolint:testableexamples
func Example_previousLine() {
	panic("xxx")
}
```
![image](https://user-images.githubusercontent.com/16486128/187074755-e76aa4eb-558a-49ae-8d4c-41d752f07b2b.png)

**Comment by ldez** [comment]
@maratori you must write nolint with the directive syntax.

The comment syntax is now deprecated.

valid:
```go
//nolint:testableexamples
```

invalid:
```go
// nolint:testableexamples
```

**Comment by maratori** [comment]
@ldez you are right. Directives are excluded form godoc.
The only exception if it's placed on the same line.
```go
//nolint:testableexamples
func Example_previousLine() {
	panic("xxx")
}

func Example_sameLine() { //nolint:testableexamples
	panic("xxx")
}
```
![image](https://user-images.githubusercontent.com/16486128/187087366-e31920bd-b95f-475a-b110-3cc89b8d259f.png)

Ok, then I'll create a linter.

**Comment by ldez** [comment]
closed by #3170
